File: backend/app/constants/categories.py
from typing import List, Dict

# Decision category definitions with semantic descriptions
CATEGORIES: Dict[str, str] = {
    "Revenue Growth": "sales revenue growth business expansion monetization income profit",
    "Maintenance": "maintenance fixing bugs technical debt infrastructure upkeep repairs",
    "Brand": "branding marketing awareness content social media presence reputation",
    "Admin": "administration meetings planning organization HR legal compliance",
    "Strategy": "strategy vision long-term planning goals direction innovation transformation",
}

CATEGORY_COLORS: Dict[str, str] = {
    "Revenue Growth": "#3B82F6",  # Blue
    "Maintenance": "#9CA3AF",     # Gray
    "Brand": "#A855F7",           # Purple
    "Admin": "#F59E0B",           # Orange
    "Strategy": "#22C55E",        # Green
}

CATEGORY_LIST: List[str] = list(CATEGORIES.keys())

File: frontend/constants/categories.ts
export const CATEGORIES = [
    "Revenue Growth",
    "Maintenance",
    "Brand",
    "Admin",
    "Strategy",
] as const;

export type Category = (typeof CATEGORIES)[number];

export const CATEGORY_DESCRIPTIONS: Record<Category, string> = {
    "Revenue Growth": "Sales, monetization, business expansion",
    Maintenance: "Bug fixes, technical debt, upkeep",
    Brand: "Marketing, content, awareness",
    Admin: "Meetings, planning, HR, legal",
    Strategy: "Vision, long-term goals, innovation",
};

export const CATEGORY_ICONS: Record<Category, string> = {
    "Revenue Growth": "📈",
    Maintenance: "🔧",
    Brand: "🎨",
    Admin: "📋",
    Strategy: "🎯",
};
